RecoTauTag/RecoTau/plugins/RecoTauBuilderConePlugin.cc

```diff
@@ -107,6 +117,63 @@ namespace xclean
   }
 }
 
+void RecoTauBuilderConePlugin::setTauQuantities(reco::PFTau& aTau,
+						double minAbsPhotonSumPt_insideSignalCone,
+						double minRelPhotonSumPt_insideSignalCone,
+						double minAbsPhotonSumPt_outsideSignalCone,
+						double minRelPhotonSumPt_outsideSignalCone) const {
+  //MB: Set tau quantities which are computed by RecoTauConstructor::get() 
+  //    method with PFRecoTauChargedHadrons not used here
+  
+  // Set charge
+  int charge = 0;
+  int leadCharge = aTau.leadPFChargedHadrCand().isNonnull() ? aTau.leadPFChargedHadrCand()->charge() : 0;
+  const std::vector<reco::PFCandidatePtr>& pfChs = aTau.signalPFChargedHadrCands();
+  for(std::vector<reco::PFCandidatePtr>::const_iterator pfCh = pfChs.begin();
+      pfCh != pfChs.end(); ++pfCh)
+    charge += (*pfCh)->charge();
+  charge = charge==0 ? leadCharge : charge;
+  aTau.setCharge(charge);
+  
+  // Set PDG id
+  aTau.setPdgId(aTau.charge() < 0 ? 15 : -15); 
+  
+  // Set Decay Mode
+  PFTau::hadronicDecayMode dm = PFTau::kNull; 
+  unsigned int nPiZeros = 0;
+  const std::vector<RecoTauPiZero>& piZeros = aTau.signalPiZeroCandidates();
+  for(std::vector<RecoTauPiZero>::const_iterator piZero = piZeros.begin();
+      piZero != piZeros.end(); ++piZero) {
+    double photonSumPt_insideSignalCone = 0.;
+    double photonSumPt_outsideSignalCone = 0.;
+      int numPhotons = piZero->numberOfDaughters();
+      for(int idxPhoton = 0; idxPhoton < numPhotons; ++idxPhoton) {
+	const reco::Candidate* photon = piZero->daughter(idxPhoton);
+	double dR = deltaR(photon->p4(), aTau.p4());
+	if(dR < aTau.signalConeSize() ){
+	  photonSumPt_insideSignalCone += photon->pt();
+	  } else {
+	    photonSumPt_outsideSignalCone += photon->pt();
+	  }
+	}
+	if( photonSumPt_insideSignalCone  > minAbsPhotonSumPt_insideSignalCone  || photonSumPt_insideSignalCone  > (minRelPhotonSumPt_insideSignalCone*aTau.pt())  ||
+	    photonSumPt_outsideSignalCone > minAbsPhotonSumPt_outsideSignalCone || photonSumPt_outsideSignalCone > (minRelPhotonSumPt_outsideSignalCone*aTau.pt()) ) ++nPiZeros;
+    }
+    // Find the maximum number of PiZeros our parameterization can hold
+    const unsigned int maxPiZeros = PFTau::kOneProngNPiZero;
+    // Determine our track index
+    unsigned int nCharged = pfChs.size();
+    unsigned int trackIndex = (nCharged - 1)*(maxPiZeros + 1);
```

Comment:
If pfChs.size()=0 this may lead to an indefinite number (as unsigned int): can it happen?